src/log_surgeon/SchemaParser.cpp

```diff
@@ -603,7 +605,7 @@ void SchemaParser::add_productions() {
     add_production(
             "Literal",
             {"Lparen", "QuestionMark", "Langle", "Identifier", "Rangle", "Regex", "Rparen"},
-            regex_capture_rule
+            std::bind(&SchemaParser::regex_capture_rule, this, std::placeholders::_1)
```

Comment:
```suggestion
            [this](NonTerminal* m) { return regex_capture_rule(m); }
```
This seems to be the morden C++ way to replace `std::bind`

Reply:
Changed.